src/varname.jl

```diff
@@ -77,26 +93,55 @@ Return the indexing tuple of the Julia variable used to generate `vn`.
 
 ```jldoctest
 julia> getindexing(@varname(x[1][2:3]))
-((1,), (2:3,))
+(@lens _[1][2:3])
 
 julia> getindexing(@varname(y))
 ()
 ```
 """
 getindexing(vn::VarName) = vn.indexing
 
+"""
+    get(obj, vn::VarName{sym})
+
+Alias for `get(obj, PropertyLens{sym}() ∘ vn.indexing)`.
```

Comment:
This will work with `SimpleVarInfo`, but not untyped `VarInfo`, and I'm not sure about typed `VarInfo`.  My original idea was to have `get(obj, vn) = get(obj, vn.indexing)`, but on a second view, this doesn't immediately seem better or worse than this variant.  Some combination will always be confusing... 

Maybe we could document the intent as: this getter is only thought for internal usage, whereas the pattern for `VarInfo` access is always with the varname as a _key_, so `getindex(vi, vn)`, which can then, if suitable, dispatch to `get(vi, vn)` or `get(vi[sym], vn.indexing)` or whatever.

Reply:
Ehhh it won't even work for `SimpleVarInfo`. `SimpleVarInfo` has a field that is a `NamedTuple` on which this would work, i.e. `svi.θ.x`.

We also use `VarName` quite a bit to index into fields of contexts, etc., e.g. in `ConditionContext` we'd use `get(context.values, vn)`, which is why I made this choice. If we want special behavior when used on `VarInfo` or `SimpleVarInfo`, this should just be done through different methods rather than overloading `get`. Which is I guess exactly what you're saying at the end there regarding `getindex(vi, vn)` :)

> Maybe we could document the intent as: this getter is only thought for internal usage

I don't see a reason why this shouldn't be used externally :confused: It's just an alias to some functionality, as it says above, that's it. We're not selling `VarName` as an implementation of a `Lens` or anything, it's just a convenient overload that applies to most of the use-cases.